sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -147,13 +150,14 @@ int ng_ipv6_netif_add_addr(kernel_pid_t pid, const ng_ipv6_addr_t *addr,
         return -ENOENT;
     }
 
-    if ((addr == NULL) || (ng_ipv6_addr_is_unspecified(addr))) {
+    if ((addr == NULL) || (ng_ipv6_addr_is_unspecified(addr)) ||
+        (prefix_len < 1) || (prefix_len > 128)) {
```

Comment:
IMHO, there is nothing wrong in helping the compiler to achieve a good level of optimization (: Nevertheless, since we are dealing with a wide range of platforms and compilers, it might make sense to manually apply some basic optimizations (like this one for range checking)